src/sagas/modelSaga.js

```diff
@@ -19,30 +22,21 @@ export function* validateModelFiles() {
     // validate the model manager
     modelManager.validateModelFiles();
 
-    yield put({
-      type: 'UPDATE_MODEL_MANAGER_SUCCEEDED',
-      modelManager,
-    });
+    yield put(actions.updateModelManagerSuccess(modelManager));
 
-    yield put({
-      type: 'UPDATE_MODEL_ERROR_SUCCEEDED',
-      error: null,
-    });
+    yield put(actions.updateModelManagerError(null));
   } catch (err) {
     err.type = 'Model';
     err.fileName = 'test.cto';
-    yield put({
-      type: 'UPDATE_MODEL_ERROR_SUCCEEDED',
-      error: err,
-    });
+    yield put(actions.updateModelManagerError(err));
   }
 }
 
+/**
+ * saga to update model file on store
```

Comment:
Maybe something along the lines of `saga which yields to updating the model file and subsequently puts a valid model in the store`